### ISSUE TITLE
Issue moving from 3.5.0 to 3.10.0 - User AdditionalData and AppRoleAssignments

### ISSUE THREAD

**Issue by chassq** [microsoftgraph/msgraph-sdk-dotnet#764]
We are looking to move from Microsoft.Graph 3.5.0 to 3.10.0. 

We are trying to access  AdditionalData (need createdDateTime see issue #669 ) and AppRoleAssignments. 

**Query currently used: (version 3.5.0 - using beta to utilize Expand method)**
`            var client = await GetServiceClient(scope, grantType, useBeta: true).ConfigureAwait(false);
            var users = await client.Users.Request()
                            .Expand("appRoleAssignments")
                            .Top(100)
                            .GetAsync().ConfigureAwait(false);`

**Query attempting to use (version 3.10.0 - using v1.0 to remove beta dependency**
`            var client = await GetServiceClient(scope, grantType).ConfigureAwait(false);
            var users = await client.Users.Request()
                            .Select(s => new { 
                                s.Id, 
                                s.Mail, 
                                s.DisplayName, 
                                s.UserPrincipalName, 
                                s.MobilePhone, 
                                s.Department, 
                                s.OfficeLocation, 
                                s.UserType,
                                s.DeletedDateTime,
                                s.AdditionalData,
                                s.AppRoleAssignments
                            })
                            .Top(100)
                            .GetAsync().ConfigureAwait(false);`

In version 3.5.0 we always get back AdditionalData and AppRoleAssignments as expected but in 3.10.0 AdditionalData and AppRoleAssignments always comes back as null. Can someone provide best/recommended practice for accessing this information as we use it to sync MS Account info with an application. We are trying to remove the dependency we have currently on the beta version of the API.

Mainly looking for guidance for using the dotnet SDK to efficiently get this information.

**Desktop (please complete the following information):**
 - OS: Windows 10 Latest Patches
- VS 2019 Latest Patches
 
[AB#7259](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/7259)

**Comment by MIchaelMainer** [comment]
You still need to expand AppRoleAssignments to request that it is returned. You don't need to selected AdditionalData as that gets populated when a property is returned when it is not defined on the object. You won't need to access AdditionalData with 3.10 as [AppRoleAssignments](https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/dev/src/Microsoft.Graph/Models/Generated/User.cs#L492) should be available on the user if they have app role assignments set on them (also assuming you expanded the property).

It will look something like this:

```c#
var client = await GetServiceClient(scope, grantType).ConfigureAwait(false); 
var users = await client.Users.Request() .Select(s => new { s.Id, s.Mail, s.DisplayName, s.UserPrincipalName, s.MobilePhone, s.Department, s.OfficeLocation, s.UserType, s.DeletedDateTime}).Expand("appRoleAssignments").Top(100).GetAsync().ConfigureAwait(false);
```

**Comment by chassq** [comment]
Hey @MIchaelMainer , thanks for the reply! One issue is that the createdDateTime is not defined on the User object (#669) and AdditionalData is still not coming back with the User object that we can see. Always seems null. Also since we still have to Expand appRoleAsignments I assume we still need to use the beta API. Yes?

**Comment by MIchaelMainer** [comment]
AppRoleAssignments is in the v1.0 API now. You no longer need the beta API for these calls. 

I missed that in my response. Since createdDateTime is still not in the schema for v1.0 but is available in the v1.0 API, you will need to make your request like:

```c#
var client = await GetServiceClient(scope, grantType).ConfigureAwait(false); 
var users = await client.Users.Request() .Select("id, mail, displayname, userPrincipalName, MobilePhone, Department, OfficeLocation, UserType, DeletedDateTime, createddatetime").Expand("appRoleAssignments").Top(100).GetAsync().ConfigureAwait(false);
```

You will need to access CreatedDateTimed from additionalData

**Comment by chassq** [comment]
Hey @MIchaelMainer thanks again! That example solved for the createdDateTime. However, with 3.10.0 we still cannot return AppRoleAssignments. We have set the Application Permissions to Directory.Read.All, AppRoleAssignment.ReadWrite.All, Directory.ReadWrite.All per the docs (https://docs.microsoft.com/en-us/graph/api/user-list-approleassignments?view=graph-rest-1.0&tabs=http) and tried the following using the v1.0 API:

**No Luck:**
` var users = await client.Users.Request()
                                .Select("id, mail, displayname, userPrincipalName, MobilePhone, Department, OfficeLocation, UserType, DeletedDateTime, AppRoleAssignments, createdDateTime")
                                .Expand("appRoleAssignments")
                                .Top(100)
                                .GetAsync()
                                .ConfigureAwait(false);`

**No Luck:**
` var users = await client.Users.Request()
                                .Select("id, mail, displayname, userPrincipalName, MobilePhone, Department, OfficeLocation, UserType, DeletedDateTime, createdDateTime")
                                .Expand("appRoleAssignments")
                                .Top(100)
                                .GetAsync()
                                .ConfigureAwait(false);`

Should we be doing something else?

**Comment by MIchaelMainer** [comment]
Hmm, that should work. Just to confirm, does it work when targeting the beta endpoint? 

Also, as an aside, the missing createdDateTime schema entry is being worked on.

**Comment by chassq** [comment]
not seeing in beta or v1.0

**Comment by MIchaelMainer** [comment]
Here's what I see:
* appRoleAssignments is defined in both the beta and v1.0 API description.
* appRoleAssignments is still only accessible via the beta endpoint. The v1.0 API description is wrong.

I can get appRoleAssignments either way with this when I target the **beta** endpoint. I also get a ton of other properties when I make these calls:

```
var result = await client.Users.Request().Expand(u => u.AppRoleAssignments).GetAsync();
var result = await client.Users.Request().Expand("approleassignments").GetAsync();
```

Approleassignments is not returned when I run this, in fact, the selected properties are not returned and the same default properties returned for a call to v1.0 are returned:

```
var result = await client.Users.Request().Expand("approleassignments").Select("id,mail,displayname,userPrincipalName,MobilePhone,Department,OfficeLocation,UserType,DeletedDateTime,createddatetime").GetAsync();
```

**Comment by MIchaelMainer** [comment]
@chassq It looks like a fix is in the pipeline. I don't have an ETA, yet.

**Comment by chassq** [comment]
Great news! Thanks @MIchaelMainer !

**Comment by andrueastman** [comment]
Closed via #788
